x-pack/plugins/security_solution/public/management/components/endpoint_execute_action/execute_action_host_response_output.tsx

```diff
@@ -99,42 +96,72 @@ interface ExecuteActionHostResponseOutputProps {
 }
 
 export const ExecuteActionHostResponseOutput = memo<ExecuteActionHostResponseOutputProps>(
-  ({ action, agentId = action.agents[0], 'data-test-subj': dataTestSubj, textSize }) => {
+  ({ action, agentId = action.agents[0], 'data-test-subj': dataTestSubj, textSize = 'xs' }) => {
+    const { canWriteExecuteOperations } = useUserPrivileges().endpointPrivileges;
+
+    const [executeOutputContent, setExecuteOutputContent] = useState<
+      undefined | ResponseActionExecuteOutputContent
+    >(undefined);
+
     const outputContent = useMemo(
       () =>
         action.outputs &&
         action.outputs[agentId] &&
         (action.outputs[agentId].content as ResponseActionExecuteOutputContent),
-      [agentId, action]
+      [action.outputs, agentId]
     );
 
+    const { data: actionDetails, isFetching } = useGetActionDetails(action.id, {
+      enabled: canWriteExecuteOperations && !outputContent,
+    });
+
+    useEffect(() => {
+      if (
+        !isFetching &&
+        actionDetails &&
+        actionDetails.data &&
+        actionDetails.data.outputs &&
+        actionDetails.data.outputs[agentId]
+      ) {
+        setExecuteOutputContent(
```

Comment:
I think you should be checking if the component is still mounted prior to calling this method.

Reply:
416154822fd34efd8e64bab195c72ff70fc9926e